virtcontainers/qemu.go

```diff
@@ -1837,3 +1849,57 @@ func (q *qemu) toGrpc() ([]byte, error) {
 
 	return json.Marshal(&qp)
 }
+
+func (q *qemu) storeState() error {
+	if q.store != nil {
+		if err := q.store.Store(store.Hypervisor, q.state); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (q *qemu) save() (s persistapi.HypervisorState) {
+	s.Pid = q.pid()
+	s.Type = string(QemuHypervisor)
+	s.UUID = q.state.UUID
+	s.HotpluggedMemory = q.state.HotpluggedMemory
+	s.HotplugVFIOOnRootBus = q.state.HotplugVFIOOnRootBus
+
+	for _, bridge := range q.state.Bridges {
+		s.Bridges = append(s.Bridges, persistapi.Bridge{
+			DeviceAddr: bridge.Address,
+			Type:       string(bridge.Type),
+			ID:         bridge.ID,
+			Addr:       bridge.Addr,
+		})
+	}
+
+	for _, cpu := range q.state.HotpluggedVCPUs {
+		s.HotpluggedVCPUs = append(s.HotpluggedVCPUs, persistapi.CPUDevice{
+			ID: cpu.ID,
+		})
+	}
+	return
+}
+
+func (q *qemu) load(s persistapi.HypervisorState) {
+	q.state.UUID = s.UUID
```

Comment:
qemu pid is not restored. Then why do we need to persist it?

Reply:
fc and ACRN use it.